cpmpy/expressions/globalconstraints.py

```diff
@@ -181,7 +181,7 @@ def decompose(self):
         return [var1 != var2 for var1, var2 in all_pairs(self.args)], []
 
     def value(self):
-        return len(set(argvals(self.args))) == len(self.args)
+        return len(set(a.value() for a in self.args)) == len(self.args)
```

Comment:
why not argvals? We switched to argvals everywhere for consistency with the partial functions